File: backend/IbuClubs.Api/Domain/Models/Activity.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace IbuClubs.Api.Domain.Models;

public class Activity
{
    [Column("activity_id")]
    public Guid ActivityId { get; set; }
    [Column("club_id")]
    public Guid ClubId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateOnly Date { get; set; }
    public TimeOnly Time { get; set; }
    [Column("activity_points")]
    public int ActivityPoints { get; set; }
    [Column("max_participants")]
    public int MaxParticipants { get; set; }

    public Club Club { get; set; }
    [NotMapped]
    public bool isEnrolled { get; set; }
    public ICollection<ActivityEnrollment> ActivityEnrollments { get; set; }
}

File: mobile/ibuclubs_mobile/lib/features/activities/domain/model/activity.dart
import 'package:ibuclubs_mobile/features/activities/domain/model/enrollment.dart';

class Activity {
  final String id;
  final String clubId;
  final String name;
  final String description;
  final String date;
  final String time; // in minutes
  final int activityPoints;
  final int maxParticipants;
  final bool isEnrolled;
  final List<Enrollment> enrollments;

  Activity({
    required this.id,
    required this.clubId,
    required this.name,
    required this.description,
    required this.date,
    required this.time,
    required this.activityPoints,
    required this.maxParticipants,
    required this.isEnrolled,
    required this.enrollments,
  });
}
